numba/core/untyped_passes.py

```diff
@@ -1452,3 +1462,42 @@ def run_pass(self, state):
         state.func_ir.blocks = new_blks
         mutated = blks != new_blks
         return mutated
+
+
+@register_pass(mutates_CFG=False, analysis_only=False)
+class ReconstructSSA(FunctionPass):
+    """Perform SSA-reconstruction
+
+    Produces minimal SSA.
+    """
+    _name = "reconstruct_ssa"
+
+    def __init__(self):
+        FunctionPass.__init__(self)
+
+    def run_pass(self, state):
+        state.func_ir = reconstruct_ssa(state.func_ir)
+        self._patch_locals(state)
+
+        # Rebuild definitions
+        state.func_ir._definitions = build_definitions(state.func_ir.blocks)
+
+        # Rerun postprocessor to update metadata
+        # example generator_info
+        post_proc = postproc.PostProcessor(state.func_ir)
+        post_proc.run(emit_dels=False)
+        return True      # XXX detect if it actually got changed
+
+    def _patch_locals(self, state):
+        # Fix locals type annotation
```

Comment:
```suggestion
        # Fix dispatcher locals dictionary type annotation
```
presume this is fixing the `@jit(locals={})` dict ?